includes/class-sensei-course.php

```diff
@@ -2929,80 +2914,102 @@ public static function the_course_enrolment_actions() {
-				// translators: Placeholder is a link to log in.
-				$message = sprintf( __( 'Or %1$s to access your purchased courses', 'woothemes-sensei' ), $login_link );
-				Sensei()->notices->add_notice( $message, 'info' );
-				Sensei_WC::the_add_to_cart_button_html( $post->ID );
-
-			} else {
-
-				if ( get_option( 'users_can_register' ) ) {
+		</section>
+		<?php
+	}
 
-					// set the permissions message
-					$anchor_before = '<a href="' . esc_url( sensei_user_login_url() ) . '" >';
-					$anchor_after  = '</a>';
-					$notice        = sprintf(
-						// translators: Placeholders are an opening and closing <a> tag linking to the login URL.
-						__( 'or %1$slog in%2$s to view this course.', 'woothemes-sensei' ),
-						$anchor_before,
-						$anchor_after
-					);
+	/**
+	 * Output the course actions like start taking course, register, etc. Note
+	 * that this expects that the user is not already taking the course; that
+	 * check is done in `the_course_enrolment_actions`.
+	 *
+	 * @access private
+	 *
+	 * @since 2.0.0
+	 */
+	public static function output_course_enrolment_actions() {
+		global $post;
 
-					self::add_course_access_permission_message( $notice );
+		$is_course_content_restricted = (bool) apply_filters( 'sensei_is_course_content_restricted', false, $post->ID );
 
-					$my_courses_page_id = '';
+		if ( is_user_logged_in() ) {
+			$should_display_start_course_form = (bool) apply_filters( 'sensei_display_start_course_form', true, $post->ID );
+			if ( $is_course_content_restricted && false == $should_display_start_course_form ) {
+				self::add_course_access_permission_message( '' );
+			}
+			if ( $should_display_start_course_form ) {
+				sensei_start_course_form( $post->ID );
+			}
+		} else {
+			if ( get_option( 'users_can_register' ) ) {
+
+				// set the permissions message
+				$anchor_before = '<a href="' . esc_url( sensei_user_login_url() ) . '" >';
+				$anchor_after  = '</a>';
+				$notice        = sprintf(
+					// translators: Placeholders are an opening and closing <a> tag linking to the login URL.
+					__( 'or %1$slog in%2$s to view this course.', 'woothemes-sensei' ),
+					$anchor_before,
+					$anchor_after
+				);
 
-					/**
-					 * Filter to force Sensei to output the default WordPress user
-					 * registration link.
-					 *
-					 * @since 1.9.0
-					 * @param bool $wp_register_link default false
-					 */
+				self::add_course_access_permission_message( $notice );
 
-					$wp_register_link = apply_filters( 'sensei_use_wp_register_link', false );
+				$my_courses_page_id = '';
 
-					$settings = Sensei()->settings->get_settings();
-					if ( isset( $settings['my_course_page'] )
-						&& 0 < intval( $settings['my_course_page'] ) ) {
+				/**
+				 * Filter to force Sensei to output the default WordPress user
+				 * registration link.
+				 *
+				 * @since 1.9.0
+				 * @param bool $wp_register_link default false
+				 */
 
```

Comment:
I'm not sure the apigen parsers like a space between here and the filter? I'm sure it is fine, though.

Reply:
Fixed!